File: rust/prime_number_checker/prime_number_checker_1.rs
// This program is designed to check if a given number is a prime number.
// A prime number is a natural number greater than 1 that has no positive divisors other than 1 and itself.
// The program will take an input number from the user and determine if it is prime.
// It will use various functions and variables to achieve this goal.

use std::io;

// Function to get user input
fn get_user_input() -> u32 {
    let mut input = String::new();
    println!("Please enter a number to check if it is prime:");
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let number: u32 = input.trim().parse().expect("Please type a number!");
    number
}

// Function to check if a number is prime
fn is_prime(number: u32) -> bool {
    if number <= 1 {
        return false;
    }
    if number == 2 {
        return true;
    }
    if number % 2 == 0 {
        return false;
    }
    let mut is_prime = true;
    let mut i = 3;
    while i * i <= number {
        if number % i == 0 {
            is_prime = false;
            break;
        }
        i += 2;
    }
    is_prime
}

// Function to print the result
fn print_result(number: u32, is_prime: bool) {
    if is_prime {
        println!("The number {} is a prime number.", number);
    } else {
        println!("The number {} is not a prime number.", number);
    }
}

// Main function
fn main() {
    let number = get_user_input();
    let prime_status = is_prime(number);
    print_result(number, prime_status);
}

File: vbnet/prime_number_checker/prime_number_checker_1.vb
' This program is a prime number checker written in Visual Basic .NET.
' The purpose of this program is to determine whether a given number is a prime number.
' A prime number is a natural number greater than 1 that has no positive divisors other than 1 and itself.
' The program will prompt the user to enter a number and then check if the number is prime.
' If the number is prime, the program will display a message indicating that the number is prime.
' If the number is not prime, the program will display a message indicating that the number is not prime.

Module PrimeNumberChecker

    Sub Main()
        ' Declare a variable to store the user's input
        Dim inputNumber As Integer

        ' Prompt the user to enter a number
        Console.WriteLine("Enter a number to check if it is prime:")
        inputNumber = Convert.ToInt32(Console.ReadLine())

        ' Call the function to check if the number is prime
        If IsPrime(inputNumber) Then
            ' If the function returns True, the number is prime
            Console.WriteLine(inputNumber & " is a prime number.")
        Else
            ' If the function returns False, the number is not prime
            Console.WriteLine(inputNumber & " is not a prime number.")
        End If

        ' Wait for the user to press a key before closing the console window
        Console.ReadKey()
    End Sub

    ' Function to check if a number is prime
    Function IsPrime(ByVal number As Integer) As Boolean
        ' A prime number is greater than 1
        If number <= 1 Then
            Return False
        End If

        ' Check for factors other than 1 and the number itself
        For i As Integer = 2 To Math.Sqrt(number)
            ' If the number is divisible by any number other than 1 and itself, it is not prime
            If number Mod i = 0 Then
                Return False
            End If
        Next

        ' If no factors are found, the number is prime
        Return True
    End Function

End Module

